app/src/main/kotlin/com/mapbox/vision/teaser/replayer/SessionsAdapter.kt

```diff
@@ -1,42 +1,180 @@
+        private set
 
     init {
-        File(basePath)
-            .listFiles()
-            .forEach { sessions.add(it.name) }
+        cameraString = context.getString(R.string.camera_text)
+        backgroundColorSelection = context.getColor(R.color.white_30_opacity)
+        backgroundColorTransparent = context.getColor(R.color.fully_transparent)
+        updateSessionsList()
     }
 
+    fun getSelectedCount() = selectedItems.size
+
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SessionViewHolder {
-        val view = LayoutInflater.from(parent.context).inflate(R.layout.row_session, parent, false)
+        val view = LayoutInflater.from(parent.context).inflate(R.layout.view_row_session, parent, false)
         return SessionViewHolder(view)
     }
 
     override fun onBindViewHolder(holder: SessionViewHolder, position: Int) {
-        holder.textView.text = sessions[position]
+        val item = items[position]
+        bindTextInfo(holder, item)
+        bindBackground(holder, item)
+        bindViewHolderClickListener(holder, item)
+        bindViewHolderLongClick(holder, item)
     }
 
-    override fun getItemCount() = sessions.size
+    private fun bindTextInfo(holder: SessionViewHolder, item: AdapterItem) {
+        holder.textSessionName.text = item.name
+        if (item is SessionItem) {
+            holder.textSessionDate.text = item.fileDate
+        } else {
+            holder.textSessionDate.text = ""
+        }
+    }
 
-    inner class SessionViewHolder(view: View) : RecyclerView.ViewHolder(view) {
-        val textView: TextView = view.findViewById(R.id.tv_session_name)
+    private fun bindBackground(holder: SessionViewHolder, item: AdapterItem) {
+        if (isMultiSelection) {
+            when {
+                item is CameraItem -> {
+                    holder.itemView.isEnabled = false
+                }
+                selectedItems.contains(item.name) -> {
+                    holder.itemView.setBackgroundColor(backgroundColorSelection)
+                    holder.iconChecked.setImageResource(R.drawable.ic_checked)
+                    holder.iconChecked.visibility = VISIBLE
+                }
+                else -> {
+                    holder.itemView.background = context.getDrawable(R.drawable.session_info_custom_ripple)
+                    holder.iconChecked.setImageResource(R.drawable.ic_not_checked)
+                    holder.iconChecked.visibility = VISIBLE
+                }
+            }
+        } else {
+            if (item is CameraItem) {
+                holder.itemView.isEnabled = true
+            } else {
+                holder.itemView.background = context.getDrawable(R.drawable.session_info_custom_ripple)
+            }
+            holder.iconChecked.visibility = INVISIBLE
+        }
+    }
+
+    private fun bindViewHolderClickListener(holder: SessionViewHolder, item: AdapterItem) {
+        if (!(item is CameraItem && isMultiSelection)) {
```

Comment:
I'd use De Morgan's law here to get rid of `(!(`

Reply:
changed, IMO the same from readability :)
but probably better for others